pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -275,7 +289,29 @@ func (c *Controller) syncService(key string) error {
 		return err
 	}
 
-	esLabelSelector := labels.Set(map[string]string{discovery.LabelServiceName: service.Name}).AsSelectorPreValidated()
+	// With the goal of different controllers being able to manage different
+	// subsets of EndpointSlices, LabelManagedBy has been added to indicate
+	// which controller or entity manages an EndpointSlice. As part of this
+	// v1.16->v1.17 change, EndpointSlices will initially be assumed to be
+	// managed by this controller unless a label is set to indicate otherwise.
+	// To ensure a seamless upgrade process, the managedBySetupAnnotation is
+	// used to indicate that LabelManagedBy has been set initially for related
+	// EndpointSlices. If it hasn't been set to the expected value here, we
+	// call setupManagedBy() to set up LabelManagedBy on each EndpointSlice.
+	// TODO(robscott): Remove this before v1.18.
+	if managedBySetup, ok := service.Annotations[managedBySetupAnnotation]; !ok || managedBySetup != managedBySetupCompleteValue {
+		err = c.setupManagedBy(service)
+		if err != nil {
+			c.eventRecorder.Eventf(service, v1.EventTypeWarning, "FailedToLabelEndpointSlices",
```

Comment:
`FailedToLabelEndpointSlices ` reason seems to be misleading

Reply:
Yeah it's hard to come up with a good but somewhat concise summary of what this is. I ended up changing this to `FailedToSetEndpointSliceManagedByLabel`, hopefully that's not too long.